SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift

```diff
@@ -29,18 +29,20 @@ class AppCenterViewController : NSViewController, NSTextFieldDelegate, NSTextVie
   @IBOutlet weak var signInButton: NSButton!
   @IBOutlet weak var signOutButton: NSButton!
   @IBOutlet weak var setLogURLButton: NSButton!
-
+  @IBOutlet weak var setAppSecretButton: NSButton!
+  
   private var dbFileDescriptor: CInt = 0
   private var dbFileSource: DispatchSourceProtocol?
 
   deinit {
-      self.dbFileSource?.cancel()
-      close(self.dbFileDescriptor)
-      UserDefaults.standard.removeObserver(self, forKeyPath: kMSStorageMaxSizeKey)
+    self.dbFileSource?.cancel()
+    close(self.dbFileDescriptor)
+    UserDefaults.standard.removeObserver(self, forKeyPath: kMSStorageMaxSizeKey)
   }
 
   override func viewWillAppear() {
     setEnabledButton?.state = appCenter.isAppCenterEnabled() ? 1 : 0
+    setAppSecretButton?.isEnabled = startUpModeForCurrentSession == 1 ? false : true
```

Comment:
You can probably introduce a constant or enum as mentioned in the iOS PR to replace the `1` value.